SystemInformer/proctree.c

```diff
@@ -1451,6 +1486,155 @@ static VOID PhpUpdateProcessNodePriorityBoost(
+    {
+        if (!ProcessNode->SvcNames)
+        {
+            ProcessNode->SvcNames = PhCreateString(L"");
+        }
+
+        if ((ProcessNode->ProcessItem->ServiceList && ProcessNode->ProcessItem->ServiceList->Count != 0))
+        {
+            PH_STRING_BUILDER stringBuilder;
+            PhInitializeStringBuilder(&stringBuilder, 1000);
+
+            // Based on itemtips.c	 :	serviceList = PhCreateList(Process->ServiceList->Count);
+            //		I don't think ServiceListLock is needed, since we are not modyfing this, just simply reading it.
+            //		In itemtips.c its inserting items
+
+            //PhAcquireQueuedLockShared(&ProcessNode->ProcessItem->ServiceListLock);
+
+            for (ULONG i = 0; i < ProcessNode->ProcessItem->ServiceList->Count; i++)
+            {
+                PPH_SERVICE_ITEM serviceItem = ProcessNode->ProcessItem->ServiceList->Items[i];
+
+                PhAppendStringBuilder(&stringBuilder, &serviceItem->Name->sr);
+                PhAppendStringBuilder2(&stringBuilder, L" (");
+                PhAppendStringBuilder(&stringBuilder, &serviceItem->DisplayName->sr);
+                PhAppendStringBuilder2(&stringBuilder, L")");
+
+                if (i + 1 < ProcessNode->ProcessItem->ServiceList->Count)
+                {
+                    PhAppendStringBuilder2(&stringBuilder, L"  |  ");
+                }
+            }
+
+            //PhReleaseQueuedLockShared(&ProcessNode->ProcessItem->ServiceListLock);
+
+            // Destroy the blank string above, before assigneding to this instead
+            PhDereferenceObject(ProcessNode->SvcNames);
+
+            ProcessNode->SvcNames = PhFinalStringBuilderString(&stringBuilder);
+        }
+
+        ProcessNode->ValidMask |= PHPN_SERVICENAMES;
+    }
+}
+
+static VOID PhpUpdateProcessServiceCount(
+    _Inout_ PPH_PROCESS_NODE ProcessNode
+)
+{
+    if (!(ProcessNode->ValidMask & PHPN_SVCCOUNT))		// Bitwise flag for calling this update method
+    {
+        ProcessNode->SvcCount = 0;     // Default to nothing
+
+        PhAcquireQueuedLockShared(&ProcessNode->ProcessItem->ServiceListLock);
+
+        if ((ProcessNode->ProcessItem->ServiceList && ProcessNode->ProcessItem->ServiceList->Count != 0))
+        {
+
+            ULONG count = ProcessNode->ProcessItem->ServiceList->Count;
+            ProcessNode->SvcCount = count;									// Remeber this ULONG count for sorting on it
+            PhPrintInt32(ProcessNode->SvcCountText, count);					// What will be displayed in the column (converting ULONG to WCHAR array (PWSTR))
+        }
+
+        PhReleaseQueuedLockShared(&ProcessNode->ProcessItem->ServiceListLock);
+
+        ProcessNode->ValidMask |= PHPN_SVCCOUNT;
+    }
+}
+
+//
+//   This reads in the newly accessible list (ProcessNode->ProcessItem->ThreadList) from procprv.c that is always storing newly createdthread TIDs
+//
+static VOID PhpUpdateProcessNodeThreadTokenCount(_Inout_ PPH_PROCESS_NODE ProcessNode)
```

Comment:
Please remove this from this PR and let's discuss then implement in another PR if it's even something we can/should accept.

Reply:
Thank you OK